client/components/content-containers/tcc-assessment-pool/edit/index.vue

```diff
@@ -0,0 +1,140 @@
+<template>
+  <div class="assessments">
+    <div class="heading">
+      <h2 class="blue-grey--text text--darken-3">Assessments</h2>
+      <v-btn v-if="hasAssessments" @click="toggleAssessments" text small>
+        {{ allSelected ? 'hide all' : 'show all' }}
+      </v-btn>
+    </div>
+    <v-alert :value="!hasAssessments" color="white" icon="mdi-information-variant">
+      Click the button below to create first assessment.
+    </v-alert>
+    <ul class="list-group">
+      <assessment-item
+        v-for="it in assessments"
+        :key="it._cid"
+        @selected="toggleSelect(it)"
+        @save="saveAssessment"
+        @delete="$emit('deleteElement', it)"
+        :assessment="it"
+        :expanded="isSelected(it)" />
+    </ul>
+    <add-element
+      @add="addAssessment"
+      :include="['ASSESSMENT']"
+      :activity="container"
+      :position="nextPosition"
+      :layout="false"
+      large
+      label="Add assessment" />
+  </div>
+</template>
+
+<script>
+import AddElement from 'tce-core/AddElement';
+import AssessmentItem from 'tce-core/AssessmentItem';
+import cuid from 'cuid';
+import filter from 'lodash/filter';
+import last from 'lodash/last';
+import map from 'lodash/map';
+import pickBy from 'lodash/pickBy';
+import sortBy from 'lodash/sortBy';
+
+export default {
+  name: 'assessment-block',
+  props: {
+    container: { type: Object, required: true },
+    tes: { type: Object, required: true }
+  },
+  data() {
+    return {
+      unsavedAssessments: {},
+      selected: [],
+      allSelected: false
+    };
+  },
+  computed: {
+    savedAssessments() {
+      return filter(this.tes, { activityId: this.container.id });
```

Comment:
Why can component get elements which are not part of this container? 🤔

Reply:
Because that is how all of the containers currently work, including the [default](https://github.com/ExtensionEngine/tailor/blob/develop/client/components/editor/structure/ContentContainers/Container.vue#L56) ones. We fetch all the content elements and then we pass all of them to each container and then each one of them filters out the ones that are theirs.

Reply:
Can we improve it now? / Is there any reasoning behind that 🤔 ?